src/backend/distributed/transaction/backend_data.c

```diff
@@ -0,0 +1,480 @@
+/*-------------------------------------------------------------------------
+ *
+ * backend_data.c
+ *
+ *  Infrastructure for managing per backend data for distributed transaction
```

Comment:
I'd remove the "distributed transaction management" bit - we'll have non-transaction related stuff in there soon. Maybe "... per backend data that can efficiently accessed by all sessions" or similar?